src/context/products/productContext.js

```diff
@@ -1,30 +1,50 @@
-import React, { useState, createContext, useContext, useEffect,useReducer } from "react";
-import { fetchProducts } from "../../services/product_services";
-import {reducer} from "./productReducer"
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+} from "react";
+import {fetchProducts,reducer,getSortedData,getFilterdData} from "./index"
```

Comment:
You don't need to mention `index` in `./index`, `./` will work fine too.

Reply:
if there will be several files in the same folder then later they will cause errors because of using this shorthand.